### PR TITLE
Update json.h

### PR DESCRIPTION
**Pull request by mrGoodWolf**
Add compact, short information about your PR for easier understanding:

- Goal of the PR
- How does the PR work?
- Does it resolve any reported issue?
- Does this relate to a goal in [the roadmap](https://github.com/minetest/minetest/blob/master/doc/direction.md)?
- If not a bug fix, why is this PR needed? What usecases does it solve?

## To do

This PR is a Work in Progress / Ready for Review.

- [ ] List
- [ ] Things
- [ ] To do

### CODE REVIEW

File: lib/jsoncpp/json/json.h [hunk 1]
```diff
@@ -167,15 +167,15 @@ template <typename T> class SecureAllocator {
   void deallocate(volatile pointer p, size_type n) {
     std::memset(p, 0, n * sizeof(T));
     // free using "global operator delete"
-    ::operator delete(p);
+    allocate::operator delete(p);
```
Comment (SwissalpS) [inline lib/jsoncpp/json/json.h:170]: does the comment in the line above still apply when you do this?

**Comment by SmallJoker** [comment]
Please propose this change upstream instead: https://github.com/open-source-parsers/jsoncpp/blob/master/include/json/allocator.h

**Comment by rubenwardy** [comment]
Hi, this is a third party library that we've included in our source, but we shouldn't modify it as it would make updating to the latest version more painful. Consider proposing this change upstream instead

File: lib/jsoncpp/json/json.h (hunk 1 above)
Comment (mrGoodWolf) [reply lib/jsoncpp/json/json.h:170]: yes

**Pull request closed without merging**